conman/routes/tests/test_models.py

```diff
@@ -295,6 +289,46 @@ def test_fall_back_to_branch(self):
         self.assertEqual(route, branch)
 
 
+class RouteGetDescendantsTest(TestCase):
+    """
+    Test Route().get_descendants().
+
+    All of these tests assert use of only one query:
+        * Get the decendants of a Route:
+
+            SELECT * FROM "routes_route"
+            WHERE (
+                NOT ("routes_route"."id" = <id>)
+                AND "routes_route"."url"::text LIKE '<url>%'
+            )
+            ORDER BY "routes_route.url" DESC
+    """
+    def test_just_created(self):
+        branch = RouteFactory.build()
+
+        with self.assertNumQueries(0):
+            # Descendants presumed nonsense as unsaved, so no query.
+            descendants = list(branch.get_descendants())
+
+        self.assertEqual(descendants, [])
+
+    def test_no_descendants(self):
+        branch = ChildRouteFactory.create()
+
+        with self.assertNumQueries(1):
+            descendants = list(branch.get_descendants())
+
+        self.assertEqual(descendants, [])
+
+    def test_descendants(self):
+        branch = ChildRouteFactory.create()
+
+        with self.assertNumQueries(1):
+            descendants = list(branch.parent.get_descendants())
+
+        self.assertEqual(descendants, [branch])
```

Comment:
The list call is to nudge the query into happening within the context manager, not for the assert. I'd rather have the second assert outside it.